tools/cli-generator/generate.go

```diff
@@ -0,0 +1,281 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"embed"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"text/template"
+	"time"
+
+	"gopkg.in/yaml.v3"
+)
+
+var (
+	//go:embed spec.yaml
+	spec []byte
+	//go:embed templates/*.txt
+	templateFolder embed.FS
+
+	templateFuncs = template.FuncMap{
+		"Year": func() int {
+			return time.Now().Year()
+		},
+		"Now": func() string {
+			return time.Now().Format(time.RFC3339)
+		},
+		"Join": strings.Join,
+	}
+
+	ErrGenerateCli = errors.New("error generating cli")
+)
+
+type CLI struct {
+	Commands []Command
+	Stores   []Store
+
+	overwrite bool
+	basePath  string
+	templates *template.Template
+}
+
+type Command struct {
+	CommandPath    string    `yaml:"command_path,omitempty"`
+	PackageName    string    `yaml:"package_name,omitempty"`
+	OutputTemplate string    `yaml:"output_template,omitempty"`
+	Description    string    `yaml:"description,omitempty"`
+	StoreName      string    `yaml:"store_name,omitempty"`
+	StoreMethod    string    `yaml:"store_method,omitempty"`
+	Template       string    `yaml:"template,omitempty"`
+	IDDescription  string    `yaml:"id_description,omitempty"`
+	IDName         string    `yaml:"id_name,omitempty"`
+	Example        string    `yaml:"example,omitempty"`
+	RequestType    string    `yaml:"request_type,omitempty"`
+	SubCommands    []Command `yaml:"sub_commands,omitempty"`
+}
+
+func (c *Command) TemplateFile() string {
+	return c.Template + ".txt"
+}
+
+func (c *Command) TemplateUnitTestFile() string {
+	return c.Template + "_test.txt"
+}
+
+type Store struct {
+	BaseFileName string          `yaml:"base_file_name,omitempty"`
+	Template     string          `yaml:"template,omitempty"`
+	Lister       *StoreInterface `yaml:"lister,omitempty"`
+	Describer    *StoreInterface `yaml:"describer,omitempty"`
+	Creator      *StoreInterface `yaml:"creator,omitempty"`
+	Updater      *StoreInterface `yaml:"updater,omitempty"`
+	Deleter      *StoreInterface `yaml:"deleter,omitempty"`
+}
+
+func (s *Store) TemplateFile() string {
+	return s.Template + ".txt"
+}
+
+func (s *Store) InterfaceNames() string {
+	names := []string{}
```

Comment:
```suggestion
	var names []string{}
```
https://github.com/mongodb/mongodb-atlas-cli/pull/1748#discussion_r1137082021 😬